diff --git a/subprojects/gst-plugins-bad/gst/rist/gstroundrobin.c b/subprojects/gst-plugins-bad/gst/rist/gstroundrobin.c
index d899e7e872..c307bafab8 100644
--- a/subprojects/gst-plugins-bad/gst/rist/gstroundrobin.c
+++ b/subprojects/gst-plugins-bad/gst/rist/gstroundrobin.c
@@ -91,15 +91,26 @@ gst_round_robin_request_pad (GstElement * element, GstPadTemplate * templ,
     const gchar * name, const GstCaps * caps)
 {
   GstPad *pad;
-
-  pad = gst_element_get_static_pad (element, name);
-  if (pad) {
-    gst_object_unref (pad);
+  GstPadDirection direction = templ->direction;
+  if (direction == GST_PAD_SRC)
+  {
+    pad = gst_pad_new_from_static_template(&src_templ, name);
+  }
+  else if (direction == GST_PAD_SINK)
+  {
+    if(GST_ELEMENT(element)->numsinkpads>0)
+    {
+      //only one sink pad allowed for this element
+      return NULL;
+    }
+    pad = gst_pad_new_from_static_template(&sink_templ, name);
+  }
+  else
+  {
     return NULL;
   }
-
-  pad = gst_pad_new_from_static_template (&src_templ, name);
-  gst_element_add_pad (element, pad);
+  
+  gst_element_add_pad(element, pad);
 
   return pad;
 }
